src/routes/auth.ts

```diff
@@ -0,0 +1,96 @@
+import express from 'express'
+import { ErrorTypes, SiweMessage } from 'siwe'
+import { validateSchema } from '../schema'
+import {
+  AuthRequestBody,
+  InvalidSiweParamsError,
+  NotImplementedError,
+} from '../types'
+import { asyncRoute } from './async-route'
+
+const MAX_EXPIRATION_TIME_MS = 4 * 60 * 60 * 1000 // 4 hours
+const VERSION = '1'
+const CHAIN_ID = 42220
+
+function validateNonce(_nonce: string) {
+  throw new NotImplementedError('Nonce validation not implemented')
+}
+
+function validateExpirationTime(issuedAt: string, expirationTime?: string) {
+  if (!expirationTime) {
+    throw new InvalidSiweParamsError('Missing expiration time')
+  }
+  const issuedAtDate = new Date(issuedAt)
+  const expirationDate = new Date(expirationTime)
+  if (
+    expirationDate.getTime() - issuedAtDate.getTime() >
+    MAX_EXPIRATION_TIME_MS
+  ) {
+    throw new InvalidSiweParamsError('Expiration time too long')
+  }
+}
+
+function validateDomainAndUri(_domain: string, _uri: string) {
+  throw new NotImplementedError('Domain and URI validation not implemented')
+}
+
+export function authRouter(): express.Router {
+  const router = express.Router()
+
+  const authRequestBodyValidator = (
+    req: express.Request,
+    _res: express.Response,
+    next: express.NextFunction,
+  ) => {
+    req.body = validateSchema<AuthRequestBody>(
+      req.body,
+      'AuthRequestBodySchema',
+    )
+    next()
+  }
+
+  router.post(
+    '/login',
+    authRequestBodyValidator,
+    asyncRoute(
+      async (
+        req: express.Request<{}, {}, AuthRequestBody>,
+        res: express.Response,
+      ) => {
+        let siweFields: SiweMessage
+
+        try {
+          const siweMessage = new SiweMessage(req.body.message)
+          siweFields = await siweMessage.validate(req.body.signature)
```

Comment:
ignorant question: does this `validate` function also handle contract wallet signatures?

Reply:
I guess the latest version of Joe's auth PR says contract account support is optional, so maybe this is fine either way..

Reply:
yes, it can do a contract wallet signature if a `provider` option is passed to the `validate` method. Although I just noticed a major PR was merged, which renames `validate` -> `verify`. https://github.com/spruceid/siwe/commit/9e79a929537046d6ecf2da59a105bd522d5ebe68.